deepmd/dpmodel/model/linear_model.py

```diff
@@ -0,0 +1,415 @@
+        elif weights == "sum":
+            self.weights = [1 for _ in range(len(models))]
+        # TODO: add more weights, for example, so-called committee models
+        elif weights == "zbl":
+            pass
+        else:
+            raise ValueError(f"Invalid weights {weights}")
+
+    def distinguish_types(self) -> bool:
+        """If distinguish different types by sorting."""
+        return all(model.distinguish_types() for model in self.models)
+
+    def get_rcut(self) -> float:
+        """Get the cut-off radius."""
+        return max(self.get_rcuts)
+
+    def get_rcuts(self) -> List[float]:
+        """Get the cut-off radius for each individual models in ascending order."""
+        return [model.get_rcut() for model in self.models]
+
+    def get_sel(self) -> List[int]:
+        return [max(self.get_sels)]
+
+    def get_sels(self) -> List[int]:
+        """Get the cut-off radius for each individual models in ascending order."""
+        return [
+            sum(model.get_sel())
+            if isinstance(model.get_sel(), list)
+            else model.get_sel()
+            for model in self.models
+        ]
+
+    def _sort_rcuts_sels(self) -> Tuple[List[int], List[float]]:
+        # sort the pair of rcut and sels in ascending order, first based on sel, then on rcut.
+        zipped = sorted(
+            zip(self.get_rcuts(), self.get_sels()), key=lambda x: (x[1], x[0])
+        )
+        return [p[0] for p in zipped], [p[1] for p in zipped]
+
+    def forward_atomic(
+        self,
+        extended_coord,
+        extended_atype,
+        nlist,
+        mapping: Optional[np.ndarray] = None,
+    ) -> Dict[str, np.ndarray]:
+        """Return atomic prediction.
+
+        Parameters
+        ----------
+        extended_coord
+            coodinates in extended region, (nframes, nall * 3)
+        extended_atype
+            atomic type in extended region, (nframes, nall)
+        nlist
+            neighbor list, (nframes, nloc, nsel).
+        mapping
+            mapps the extended indices to local indices.
+
+        Returns
+        -------
+        result_dict
+            the result dict, defined by the fitting net output def.
+        """
+        nframes, nloc, nnei = nlist.shape
+        extended_coord = extended_coord.reshape(nframes, -1, 3)
+        sorted_rcuts, sorted_sels = self._sort_rcuts_sels()
+        nlists = build_multiple_neighbor_list(
+            extended_coord,
+            nlist,
+            sorted_rcuts,
+            sorted_sels,
+        )
+        nlists_ = [
+            nlists[str(rcut) + "_" + str(sel)]
+            for rcut, sel in zip(self.get_rcuts(), self.get_sels())
+        ]
+        ener_list = [
+            model.forward_atomic(
+                extended_coord,
+                extended_atype,
+                nlist,
+                mapping,
+            )["energy"]
+            for model, nlist in zip(self.models, nlists_)
+        ]
```

Comment:
We can potentially reuse some of the code here in the ZBLModel, will refactor in a Future PR, when we finalize the bias term.